### DISCUSSION TITLE
Navigation timeout when running tests in Jenkins

### DISCUSSION THREAD

**Discussion by TilakShrma** [getgauge/taiko#1941] [General]
Hi, 
We are running Gauge + Taiko tests as a part of continuous deployment in Jenkins, but almost every time we are getting Navigation timeout error even though in generated reports it is visible that target page is completely loaded. 
Same issue is not reproducible when running the tests locally.

Below are the environment details: 
```
Gauge version: 1.1.1
Commit Hash: 6e0d83f
Plugins
js (2.3.12)
Taiko version: 1.2.2
```

goto step implementation 

`await goto(url.href, {waitForEvents: ['loadEventFired']});`

We have also set navigation timeout in setConfig:

```
beforeSuite(

    async () => {
        setConfig({ navigationTimeout: 120000 });
    }
);
```

Is there any way to debug what is causing different behavior in Jenkins ? Or any possible solution for this ?

**Comment by sriv** [comment]
Couple of observations:

- you are using an old version of gauge, please try to upgrade (1.1.7 is the current version)
- the `setConfig` needs an `await` in the beforesuite. 

that said, the navigationTimeout error can occur even when the page seems loaded, if it is waiting for some resource to load. this could be an image or a javacript file etc.

**Reply by TilakShrma** [reply]
@sriv , Thanks for your reply. 

On the part of page getting completely loaded, i noticed that there were some pieces that were not loaded completely. 

I tried with :

- Updated gauge (v1.1.7) and taiko (v1.2.2)
- Added await in `setConfig` 

Still i am observing same behavior. 

Is there any way to debug this in Jenkins what resource might be blocking the execution, or to log occurrence of these page events ('loadEventFired', 'firstContentfulPaint' etc )?
